aptos-move/framework/src/natives/aggregator_natives/helpers_v2.rs

```diff
@@ -14,16 +14,20 @@ use move_vm_types::values::{StructRef, Value};
 const VALUE_FIELD_INDEX: usize = 0;
 const LIMIT_FIELD_INDEX: usize = 1;
 
-pub(crate) fn aggregator_value_field_as_id(value: u128) -> PartialVMResult<DelayedFieldID> {
-    u128_to_u64(value).map(DelayedFieldID::new)
+pub(crate) fn aggregator_value_field_as_id(value: u128, resolver: &dyn DelayedFieldResolver) -> PartialVMResult<DelayedFieldID> {
+    let value_u64 = u128_to_u64(value)?;
+    Ok(resolver.validate_and_convert_delayed_field_id(value_u64)?)
```

Comment:
So here PanicError gets converted to invariant violation, but the state is not recovered. I guess we cannot fall-back from here anyway?

Reply:
no, it gets converted to DELAYED_FIELDS_CODE_INVARIANT_ERROR